onnxruntime/core/mlas/lib/qgemm_kernel_smmla.cpp

```diff
@@ -395,7 +395,7 @@ MlasGemmQuantCopyPackA<MLAS_GEMM_S8S8_KERNEL_SMMLA>(
         const int8_t* a2 = a1 + lda;
         const int8_t* a3 = a2 + lda;
 
-        size_t k = CountK;
+        int k = static_cast<int>(CountK);
```

Comment:
Why do we need to do this? The conversion is not free.

Reply:
we have same usage for CountM too.

Reply:
Hi @yufenglee , thanks for the review, good question. for `CountM` we don't need the `int` range because it is decremented conditionally (for eg: CountM -= 2 only if CountM >2) whereas the loop counter `k` is handled differently. Following is one of the snippets where the `size_t` type can get into an infinite loop, there are several other places we have similar logic and we need `int` range for this.

btw, this PR is only for smmla and ummla kernels. I think we need similar fix for dot kernels as well.

https://github.com/microsoft/onnxruntime/blob/main/onnxruntime/core/mlas/lib/qgemm_kernel_smmla.cpp#L630
```
            while (k > 0) {
                d[0] = *a0++;
                d[8] = *a1++;

                d += 1;
                k -= 1;
            }
```

Reply:
Hi @yufenglee , I think you're right, we don't need this as both `CountM` and `k` are decremented conditionally. for example,  `CountM -= 2 only when CountM > 2` and` k -= 1 only when k > 0` so the `size_t` range is good enough. I will close it.